src/runtime-rs/crates/resource/src/volume/share_fs_volume.rs

```diff
@@ -59,36 +68,91 @@ impl ShareFsVolume {
                 }
             }
             Some(share_fs) => {
+                let readonly = m.options.iter().any(|opt| opt == "ro");
+
                 let share_fs_mount = share_fs.get_share_fs_mount();
-                let mount_result = share_fs_mount
-                    .share_volume(ShareFsVolumeConfig {
-                        cid: cid.to_string(),
-                        source: m.source.clone(),
-                        target: file_name,
-                        readonly: m.options.iter().any(|o| *o == "ro"),
-                        mount_options: m.options.clone(),
-                        mount: m.clone(),
-                        is_rafs: false,
+                if let Some(mut mounted_info) = share_fs.get_mounted_info(&m.source).await {
+                    // Mounted at least once
+                    let guest_path = mounted_info
+                        .guest_path
+                        .clone()
+                        .as_os_str()
+                        .to_str()
+                        .unwrap()
+                        .to_owned();
+                    if !readonly && mounted_info.readonly() {
+                        // The current mount should be upgraded to readwrite permission
+                        info!(
+                            sl!(),
+                            "The mount will be upgraded, mount = {:?}, cid = {}", m, cid
+                        );
+                        share_fs_mount
+                            .upgrade(
+                                &mounted_info
+                                    .file_name()
+                                    .context("get name of mounted info")?,
+                            )
+                            .await
+                            .context("upgrade mount")?;
+                    }
+                    if readonly {
+                        mounted_info.ro_ref_count += 1;
+                    } else {
+                        mounted_info.rw_ref_count += 1;
+                    }
+                    share_fs
+                        .set_mounted_info(&m.source, mounted_info)
```

Comment:
Do we need any lock when manipulating mounted_info? There are a whole bunch of race windows here.

Reply:
IMO, we don't need an extra lock for `mounted_info`, because `ResourceManager::inner` whose type is `Arc<RwLock<ResourceManagerInner>>` has a RwLock which ensures we can manipulate it safely.

Reply:
Yes, you are right. I am working on it.